staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/strategy.go

```diff
@@ -40,14 +43,15 @@ type CustomResourceDefinitionStorageStrategy struct {
 	validator       customResourceValidator
 }
 
-func NewStrategy(typer runtime.ObjectTyper, namespaceScoped bool, kind schema.GroupVersionKind) CustomResourceDefinitionStorageStrategy {
+func NewStrategy(typer runtime.ObjectTyper, namespaceScoped bool, kind schema.GroupVersionKind, crd apiextensions.CustomResourceDefinition) CustomResourceDefinitionStorageStrategy {
```

Comment:
better use a pointer for `crd`. With nested structures ownership is not obvious anyway. We won't modify `crd` either, will we? So no need for a deepcopy either.